lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -8850,50 +8851,28 @@ void EmitBinaryReference(ParseNode *pnode1, ParseNode *pnode2, ByteCodeGenerator
     Emit(pnode2, byteCodeGenerator, funcInfo, false);
 }
 
-void EmitUseBeforeDeclarationRuntimeError(ByteCodeGenerator * byteCodeGenerator, Js::RegSlot location, bool fLoadUndef)
+void EmitUseBeforeDeclarationRuntimeError(ByteCodeGenerator * byteCodeGenerator, Js::RegSlot location)
 {
     byteCodeGenerator->Writer()->W1(Js::OpCode::RuntimeReferenceError, SCODE_CODE(JSERR_UseBeforeDeclaration));
-    // Load something into register in order to do not confuse IRBuilder. This value will never be used.
-    if (fLoadUndef)
+
+    if (location != Js::Constants::NoRegister)
     {
+        // Optionally load something into register in order to do not confuse IRBuilder. This value will never be used.
         byteCodeGenerator->Writer()->Reg1(Js::OpCode::LdUndef, location);
     }
 }
 
-bool EmitUseBeforeDeclaration(ParseNode *pnode, ByteCodeGenerator *byteCodeGenerator, FuncInfo *funcInfo)
+bool EmitUseBeforeDeclaration(Symbol *sym, ByteCodeGenerator *byteCodeGenerator, FuncInfo *funcInfo)
```

Comment:
True! I will make it void.

`EmitUseBeforeDeclaration` is now only used in one location and I couldn't reason about it to sufficiently convince myself that I could reduce down that call site to a direct call to `EmitUseBeforeDeclarationRuntimeError`.  In particular the `HasStaticPathToAncestor` guard had me concerned.

Instead of inlining this function at that one call site I decided to leave it as a proper function.

I would be thrilled if we can reduce this down further and eliminate it altogether at some point.

---
In reply to: [62568830](https://github.com/Microsoft/ChakraCore/pull/936#discussion_r62568830) [](ancestors = 62568830)